packages/twenty-front/src/modules/object-record/record-index/options/components/RecordIndexOptionsDropdownContent.tsx

```diff
@@ -99,6 +74,41 @@ export const TableOptionsDropdownContent = ({
     TableOptionsHotkeyScope.Dropdown,
   );
 
+  const {
+    handleColumnVisibilityChange,
+    handleReorderColumns,
+    visibleTableColumns,
+    hiddenTableColumns,
+  } = useRecordIndexOptionsForTable(recordIndexId);
+
+  const {
+    visibleBoardFields,
+    hiddenBoardFields,
+    handleReorderBoardFields,
+    handleBoardFieldVisibilityChange,
+  } = useRecordIndexOptionsForBoard({
+    objectNameSingular,
+    viewBarId: recordIndexId,
+  });
+
+  const visibleRecordFields =
```

Comment:
right now we have this distinction because the Table is handling the re-ordering and visibility change of column itself. I think both board and table should share this logic and this logic should be within record index. The only doubt I have is because of column resizing which is specific to the table.

Let's revisit this when we refactor the viewBar